cpp/daal/src/algorithms/service_kernel_math.h

```diff
@@ -531,6 +531,110 @@ class ChebyshevDistances : public PairwiseDistances<FPType, cpu>
     const NumericTable & _b;
 };
 
+#if defined(__INTEL_COMPILER)
+
+template <>
+float MinkowskiDistances<float, avx512>::computeDistance(const float * x, const float * y, const size_t n)
+{
+    daal::internal::mkl::MklMath<float, avx512> math;
+
+    const size_t vecSize   = 16;
+    float d                = 0.0;
+    const size_t nBlocks   = n / vecSize;
+    float * tmp            = new float[vecSize];
+    const __m512 * ptr512x = (__m512 *)x;
+    const __m512 * ptr512y = (__m512 *)y;
+
+    if (_p == 1.0)
+    {
+        size_t i = 0;
+        for (; i < nBlocks; ++i)
+        {
+            d += _mm512_reduce_add_ps(_mm512_abs_ps(_mm512_sub_ps(ptr512x[i], ptr512y[i])));
```

Comment:
As next step please try using simple _mm512_add_ps in loop into __mm512 variable , and make only one horizontal reduce after the loop.